src/StoreKeeper/WooCommerce/B2C/Endpoints/Webhooks/EventsHandler.php

```diff
@@ -255,6 +270,106 @@ private function processFullSyncEvents($events)
         }
     }
 
+    private function processProductOnlyEvents($events)
+    {
+        if (!is_array($events)) {
+            return;
+        }
+
+        $taskData = ['storekeeper_id' => (int) $this->getId()];
+        $storekeeper_id = (int) $this->getId();
+        $categoryType = $this->getCategoryType($storekeeper_id) ?? $this->getCategoryTypeFromWordpress($storekeeper_id);
+        $module = $this->getModule();
+
+        foreach ($events as $id => $event) {
```

Comment:
That's really a lot of duplicate code. split into functions and call them inside the `processXXXEvents`